swatch-core/src/main/java/org/candlepin/subscriptions/util/ClockConfiguration.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.util;
+
+import org.candlepin.clock.ApplicationClock;
+import org.springframework.boot.autoconfigure.AutoConfigurationExcludeFilter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.TypeExcludeFilter;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.FilterType;
+
+/** Provides access to common utility beans. */
+@Configuration
```

Comment:
Do we want to use the `@AutoConfiguration` annotation here?  Also if we want it to be picked up with auto-configuration, we'll need to [add the fully qualified class name](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#features.developing-auto-configuration.locating-auto-configuration-candidates) to `META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports`.

Reply:
@Configuration and @Autoconfiguration work the same. Actually, META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports is already there and works as expected.

Reply:
There is a slight difference

``` 
Auto-configuration classes are regular {@link Configuration @Configuration} with the exception that
{@literal Configuration#proxyBeanMethods() proxyBeanMethods} is always {@code false}.
```

but more importantly, using `@AutoConfiguration` indicates to future developers that this class isn't just a regular configuration.

Reply:
> Actually, META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports is already there and works as expected

I missed that in the review.  It's in `swatch-core` though.  Does it need to be in `swatch-common-clock`?

Reply:
swatch-common-clock is framework agnostic, that's why.

Reply:
> [swatch-common-clock](https://issues.redhat.com/browse/SWATCH-common-clock) is framework agnostic, that's why.

Makes sense.  I would still like to annotate the configuration with `@AutoConfiguration` though.

Reply:
Updated